csi/src/nodeplugin_grpc.rs

```diff
@@ -0,0 +1,100 @@
+//! the CSI framework. This service must be deployed on all nodes the the
+//! Mayastor CSI node plugin is deployed.
+use crate::freezefs;
+use freezefs::{freeze_volume, unfreeze_volume, FreezeFsError};
+use mayastor_node_plugin::*;
+use tonic::{transport::Server, Code, Request, Response, Status};
+
+pub mod mayastor_node_plugin {
+    tonic::include_proto!("mayastornodeplugin");
+}
+
+#[derive(Debug, Default)]
+pub struct MayastorNodePluginSvc {}
+
+impl From<FreezeFsError> for Status {
+    fn from(err: FreezeFsError) -> Self {
+        let msg = format!("{:?}", err);
+        match err {
+            FreezeFsError::VolumeNotFound {
+                volid: _,
+            } => Status::new(Code::NotFound, msg),
+            FreezeFsError::FsfreezeFailed {
+                volid: _,
+                error: _,
+            } => Status::new(Code::Internal, msg),
+            FreezeFsError::InvalidVolumeId {
+                volid: _,
+                source: _,
+            } => Status::new(Code::InvalidArgument, msg),
+            FreezeFsError::InternalFailure {
+                volid: _,
+                source: _,
+            } => Status::new(Code::Internal, msg),
+            FreezeFsError::IOError {
+                volid: _,
+                source: _,
+            } => Status::new(Code::Unknown, msg),
+        }
+    }
+}
+
+#[tonic::async_trait]
+impl mayastor_node_plugin_server::MayastorNodePlugin for MayastorNodePluginSvc {
+    async fn freeze_fs(
+        &self,
+        request: Request<FreezeFsRequest>,
+    ) -> Result<Response<FreezeFsReply>, Status> {
+        let volume_id = request.into_inner().volume_id;
+        debug!("freeze_fs({})", volume_id);
+        match freeze_volume(&volume_id).await {
+            Ok(_) => Ok(Response::new(mayastor_node_plugin::FreezeFsReply {})),
+            Err(e) => Err(Status::from(e)),
+        }
+    }
+
+    async fn unfreeze_fs(
+        &self,
+        request: Request<UnfreezeFsRequest>,
+    ) -> Result<Response<UnfreezeFsReply>, Status> {
+        let volume_id = request.into_inner().volume_id;
+        debug!("unfreeze_fs({})", volume_id);
+        match unfreeze_volume(&volume_id).await {
+            Ok(_) => {
+                Ok(Response::new(mayastor_node_plugin::UnfreezeFsReply {}))
+            }
+            Err(e) => Err(Status::from(e)),
+        }
+    }
+}
+
+pub struct MayastorNodePluginGrpcServer {}
+
+impl MayastorNodePluginGrpcServer {
+    pub async fn run(endpoint: std::net::SocketAddr) -> Result<(), ()> {
+        info!(
+            "Mayastor node plugin gRPC server configured at address {:?}",
+            endpoint
+        );
+        let svc = Server::builder()
+            .add_service(
+                mayastor_node_plugin_server::MayastorNodePluginServer::new(
+                    MayastorNodePluginSvc {},
+                ),
+            )
+            .serve(endpoint);
+        match svc.await {
```

Comment:
This seems like a copy-paste from the mayastor service as the match svc.await is unless. We can simply do svc.await; or even remove the svc. This gets rid of a useless match here.

Reply:
Done.